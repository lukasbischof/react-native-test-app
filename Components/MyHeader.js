import React, { Component } from 'react';
import { Text } from 'react-native';

export default class MyHeader extends React.Component {
  render() {
    let titleStyle = {
      fontSize: 40,
      marginTop: 20,
      marginBottom: 20,
      textAlign: 'center'
    };

    return (
      <Text style={titleStyle}>{this.props.content}</Text>
    );
  }
}
