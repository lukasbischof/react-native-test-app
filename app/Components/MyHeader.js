import React, { Component } from 'react';
import { Text, View } from 'react-native';

export default class MyHeader extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let titleStyle = {
      fontSize: 40,
      marginTop: 20,
      marginBottom: 20,
      textAlign: 'center'
    };

    return (
      <View testID={this.props.testID}>
        <Text testID="myheadertext" style={titleStyle}>{this.props.content}</Text>
      </View>
    );
  }
}
