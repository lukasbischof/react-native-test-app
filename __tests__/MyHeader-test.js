import 'react-native';
import React from 'react';
import MyHeader from '../Components/MyHeader';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<MyHeader content="Hello" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders', () => {
  const myHeaderTree = renderer.create(<MyHeader />);
  expect(myHeaderTree).toBeTruthy();
});
