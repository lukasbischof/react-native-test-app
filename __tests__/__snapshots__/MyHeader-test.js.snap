// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontSize": 40,
      "marginTop": 20,
      "textAlign": "center",
    }
  }
>
  Hello
</Text>
`;
